Iter18: add docs

mp1947 opened the pull request:
1. добавлена документация

capsulated reviewed (changes requested):
- тестов совсем мало, 10% ((( надо бы добавить
- example_test.go тоже совсем минималистичный, все или хотябы самые основные эндпоинты нужно описать
- ✅ godoc супер, только надо пару пакетов вытащить в корень, чтобы импортируемые были: middleware и сжатие - точно подходящие: то, что можно импортировать, например, в выпускном

mp1947 commented:
> * тестов совсем мало, 10% ((( надо бы добавить

добавил тестов для всех основных методов/хэндлеров.
также добавил комманду coverage для make (без учета internal/mock,repository/database,repository/inmemory, тк тестирую сам repository через service и моки бд)

```
make coverage
go test -covermode=count -coverprofile=coverage.out ./...
?       github.com/mp1947/ya-url-shortener      [no test files]
        github.com/mp1947/ya-url-shortener/cmd/shortener                coverage: 0.0% of statements
ok      github.com/mp1947/ya-url-shortener/config       0.286s  coverage: 76.0% of statements
ok      github.com/mp1947/ya-url-shortener/internal/auth        0.464s  coverage: 72.2% of statements
?       github.com/mp1947/ya-url-shortener/internal/dto [no test files]
?       github.com/mp1947/ya-url-shortener/internal/errors      [no test files]
ok      github.com/mp1947/ya-url-shortener/internal/eventlog    0.648s  coverage: 83.3% of statements
ok      github.com/mp1947/ya-url-shortener/internal/handler     0.921s  coverage: 62.8% of statements
ok      github.com/mp1947/ya-url-shortener/internal/logger      0.706s  coverage: 88.9% of statements
ok      github.com/mp1947/ya-url-shortener/internal/middleware  1.094s  coverage: 93.8% of statements
        github.com/mp1947/ya-url-shortener/internal/mocks               coverage: 0.0% of statements
?       github.com/mp1947/ya-url-shortener/internal/model       [no test files]
ok      github.com/mp1947/ya-url-shortener/internal/repository  1.041s  coverage: 59.1% of statements
        github.com/mp1947/ya-url-shortener/internal/repository/database         coverage: 0.0% of statements
        github.com/mp1947/ya-url-shortener/internal/repository/inmemory         coverage: 0.0% of statements
ok      github.com/mp1947/ya-url-shortener/internal/router      1.154s  coverage: 94.1% of statements
ok      github.com/mp1947/ya-url-shortener/internal/service     0.957s  coverage: 81.1% of statements
ok      github.com/mp1947/ya-url-shortener/internal/usecase     1.302s  coverage: 83.3% of statements
ok      github.com/mp1947/ya-url-shortener/pkg/gzip     1.305s  coverage: 100.0% of statements
        github.com/mp1947/ya-url-shortener/pkg/middleware               coverage: 0.0% of statements
grep -vE "mocks|repository/database|repository/inmemory" coverage.out > coverage.cleaned.out
go tool cover -func=coverage.cleaned.out
github.com/mp1947/ya-url-shortener/cmd/shortener/main.go:17:                    main                    0.0%
github.com/mp1947/ya-url-shortener/config/config.go:34:                         InitConfig              76.0%
github.com/mp1947/ya-url-shortener/internal/auth/auth.go:21:                    CreateToken             80.0%
github.com/mp1947/ya-url-shortener/internal/auth/auth.go:36:                    Validate                69.2%
github.com/mp1947/ya-url-shortener/internal/eventlog/eventlog.go:30:            NewEventProcessor       75.0%
github.com/mp1947/ya-url-shortener/internal/eventlog/eventlog.go:49:            WriteEvent              100.0%
github.com/mp1947/ya-url-shortener/internal/eventlog/eventlog.go:56:            IncrementUUID           100.0%
github.com/mp1947/ya-url-shortener/internal/handler/delete_user_urls.go:23:     DeleteUserURLs          69.2%
github.com/mp1947/ya-url-shortener/internal/handler/get_original_by_id.go:22:   GetOriginalURLByID      62.5%
github.com/mp1947/ya-url-shortener/internal/handler/get_user_urls.go:27:        GetUserURLs             61.5%
github.com/mp1947/ya-url-shortener/internal/handler/ping.go:24:                 Ping                    0.0%
github.com/mp1947/ya-url-shortener/internal/handler/shorten.go:30:              ShortenURL              75.0%
github.com/mp1947/ya-url-shortener/internal/handler/shorten_batch.go:28:        BatchShortenURL         69.2%
github.com/mp1947/ya-url-shortener/internal/handler/shorten_json.go:32:         JSONShortenURL          64.7%
github.com/mp1947/ya-url-shortener/internal/logger/logger.go:14:                InitLogger              88.9%
github.com/mp1947/ya-url-shortener/internal/middleware/auth.go:17:              AuthMiddleware          93.8%
github.com/mp1947/ya-url-shortener/internal/repository/repository.go:34:        CreateRepository        59.1%
github.com/mp1947/ya-url-shortener/internal/router/router.go:22:                CreateRouter            94.1%
github.com/mp1947/ya-url-shortener/internal/service/delete_batch.go:18:         DeleteURLsBatch         100.0%
github.com/mp1947/ya-url-shortener/internal/service/delete_batch.go:34:         ProcessDeletions        80.0%
github.com/mp1947/ya-url-shortener/internal/service/get_original_url.go:13:     GetOriginalURL          71.4%
github.com/mp1947/ya-url-shortener/internal/service/get_user_urls.go:22:        GetUserURLs             77.8%
github.com/mp1947/ya-url-shortener/internal/service/helpers.go:7:               generateShortURL        100.0%
github.com/mp1947/ya-url-shortener/internal/service/shorten_url.go:26:          ShortenURL              81.8%
github.com/mp1947/ya-url-shortener/internal/service/shorten_url_batch.go:26:    ShortenURLBatch         84.6%
github.com/mp1947/ya-url-shortener/internal/usecase/id_generator.go:19:         GenerateIDFromURL       83.3%
github.com/mp1947/ya-url-shortener/pkg/gzip/gzip.go:20:                         Write                   100.0%
github.com/mp1947/ya-url-shortener/pkg/gzip/gzip.go:26:                         WriteHeader             100.0%
github.com/mp1947/ya-url-shortener/pkg/middleware/middleware.go:19:             LoggerMiddleware        0.0%
github.com/mp1947/ya-url-shortener/pkg/middleware/middleware.go:49:             GzipMiddleware          0.0%
github.com/mp1947/ya-url-shortener/pkg/middleware/middleware.go:101:            shouldUseGzip           0.0%
total:                                                                          (statements)            58.9%
```
> * example_test.go тоже совсем минималистичный, все или хотябы самые основные эндпоинты нужно описать

добавил example_test для всех хэндлеров: 59402f8ac1f71ff5c9220076a82c0650dd119706

> * ✅ godoc супер, только надо пару пакетов вытащить в корень, чтобы импортируемые были: middleware и сжатие - точно подходящие: то, что можно импортировать, например, в выпускном

перенес middleware и gzip (кроме auth) в pkg: https://github.com/mp1947/ya-url-shortener/pull/18/commits/189fcc96acdcc7450120534bc62fd5eaccca82bb

capsulated reviewed (approved):
Супер! Всё круто!
